nodejs/lib/inference-session.ts

```diff
@@ -0,0 +1,242 @@
+   *
+   * REMARK: different from input argument, in output, OnnxValue is optional. If an OnnxValue is present it will be
+   * used as a pre-allocated value by the inference engine; if omitted, inference engine will allocate buffer
+   * internally.
+   */
+  type FetchesType = readonly string[]|NullableOnnxValueMapType;
+
+  type ReturnType = {readonly [name: string]: OnnxValue};
+
+  //#endregion
+
+  //#region session options
+
+  /**
+   * A set of configurations for session behavior.
+   */
+  export interface SessionOptions {
+    /**
+     * An array of execution provider options.
+     *
+     * An execution provider option can be a string indicating the name of the execution provider,
+     * or an object of corresponding type.
+     */
+    executionProviders?: readonly SessionOptions.ExecutionProviderConfig[];
+
+    /**
+     * The intra OP threads number.
+     */
+    intraOpNumThreads?: number;
+
+    /**
+     * The inter OP threads number.
+     */
+    interOpNumThreads?: number;
+
+    /**
+     * The optimization level.
+     */
+    graphOptimizationLevel?: 'disabled'|'basic'|'extended'|'all';
+
+    /**
+     * Whether enable CPU memory arena.
+     */
+    enableCpuMemArena?: boolean;
+
+    /**
+     * Whether enable memory pattern.
+     */
+    enableMemPattern?: boolean;
+
+    /**
+     * Execution mode.
+     */
+    executionMode?: 'sequential'|'parallel';
+
+    /**
+     * Log ID.
+     */
+    logId?: string;
+
+    /**
+     * Log severity level. See
+     * https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/common/logging/severity.h
+     */
+    logSeverityLevel?: 0|1|2|3|4;
+  }
+
+  export namespace SessionOptions {
+    //#region execution providers
+
+    // currently we only have CPU and CUDA EP support. in future to support more.
+    interface ExecutionProviderOptionMap {
+      cpu: CpuExecutionProviderOption;
+      cuda: CudaExecutionProviderOption;
+    }
+
+    type ExecutionProviderName = keyof ExecutionProviderOptionMap;
+    type ExecutionProviderConfig = ExecutionProviderOptionMap[ExecutionProviderName]|ExecutionProviderName;
+
+    export interface CpuExecutionProviderOption {
+      readonly name: 'cpu';
+      useArena?: boolean;
+    }
+    export interface CudaExecutionProviderOption {
+      readonly name: 'cuda';
+      deviceId?: number;
+    }
+    //#endregion
+  }
+
+  //#endregion
+
+  //#region run options
+
+  /**
+   * A set of configurations for inference run behavior
+   */
+  export interface RunOptions {
+    /**
+     * Log severity level. See
+     * https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/common/logging/severity.h
+     */
+    logSeverityLevel?: 0|1|2|3|4;
+
+    /**
+     * A tag for the Run() calls using this
+     */
+    tag?: string;
+  }
+
+  //#endregion
+
+  //#region value metadata
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface
+  interface ValueMetadata {
+    // TBD
```

Comment:
Just curious - what exactly is this a placeholder for ?

Reply:
We have interfaces in C-API as well as C# API to get some meta-data from a model. This meta-data may contains information about inputs/outputs, for example, their dims (maybe a fixed number, a name or omit) and type. those information may help users.

For now, however, I haven't decide what to put into a ValueMetadata. this may need a futher dicsussion before next release